### PR TITLE
merge in from develop

### PR DESCRIPTION
**Pull request by ameijer**
## What does this PR change?
* fixes self code review fixes 

## Does this PR relate to any other PRs?
* 

## How will this PR impact users?
* 

## Does this PR address any GitHub or Zendesk issues?
* Closes ...

## How was this PR tested?
* 

## Does this PR require changes to documentation?
* 

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

### CODE REVIEW

**Review by nik-kc — CHANGES_REQUESTED** [review]
Looks good! Though now that we've removed the nil-response/nil-error pattern, we'll need to update [this conditional](https://github.com/ameijer/opencost/blob/atm/working-branch-fixes/pkg/customcost/repositoryquerier.go#L51) to check for empty responses.

**Comment by ameijer** [comment]
thanks nik, fixed, can you take another look?

File: pkg/customcost/pipelineservice.go [hunk 1]
```diff
@@ -199,8 +199,11 @@ func (s *PipelineService) GetCustomCostRebuildHandler() func(w http.ResponseWrit
 func (s *PipelineService) GetCustomCostStatusHandler() func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 	// If Reporting Service is nil, always return 501
 	if s == nil {
+		resultStatus := Status{
+			Enabled: false,
```
Comment (nik-kc) [inline pkg/customcost/pipelineservice.go:203]: I like the explicit `false`, even though that's the zero value. Makes it more obvious what's happening here.

File: pkg/customcost/pipelineservice.go [hunk 2]
```diff
@@ -199,8 +199,11 @@ func (s *PipelineService) GetCustomCostRebuildHandler() func(w http.ResponseWrit
 func (s *PipelineService) GetCustomCostStatusHandler() func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 	// If Reporting Service is nil, always return 501
```
Comment (nik-kc) [inline pkg/customcost/pipelineservice.go]: I hate to say it, but `Reporting` should be `Pipeline`.

File: pkg/customcost/repositoryquerier.go [hunk 3]
```diff
@@ -48,7 +48,7 @@ func (rq *RepositoryQuerier) QueryTotal(ctx context.Context, request CostTotalRe
 			ccResponse, err := repo.Get(queryStart, domain)
 			if err != nil {
 				return nil, fmt.Errorf("QueryTotal: %w", err)
-			} else if ccResponse == nil {
+			} else if ccResponse == nil || ccResponse.Start == nil || ccResponse.End == nil {
```
Comment (nik-kc) [inline pkg/customcost/repositoryquerier.go:51]: :100:

File: pkg/costmodel/router.go [hunk 4]
```diff
@@ -1846,8 +1846,12 @@ func Initialize(additionalConfigWatchers ...*watcher.ConfigMapWatcher) *Accesses
...
 	a.Router.GET("/cloud/config/delete", a.CloudConfigController.GetDeleteConfigHandler())
 
-	a.Router.GET("/customCost/total", a.CustomCostQueryService.GetCustomCostTotalHandler())
-	a.Router.GET("/customCost/timeseries", a.CustomCostQueryService.GetCustomCostTimeseriesHandler())
+	if env.IsCustomCostEnabled() {
+		a.Router.GET("/customCost/total", a.CustomCostQueryService.GetCustomCostTotalHandler())
+		a.Router.GET("/customCost/timeseries", a.CustomCostQueryService.GetCustomCostTimeseriesHandler())
+	}
+
+	a.Router.GET("/customCost/status", a.CustomCostPipelineService.GetCustomCostStatusHandler())
```
Comment (nik-kc) [inline pkg/costmodel/router.go:1854]: `CustomCostPipelineService` is going to be `nil` here if custom costs are disabled 😞

Comment (nik-kc) [reply pkg/costmodel/router.go:1854]: Disregard!

**Review by nik-kc — APPROVED** [review]

**Review by teevans — APPROVED** [review]

**Pull request merged**